src/logger.service.ts

```diff
@@ -23,148 +17,141 @@ const Levels = [
-        level: Levels[level],
-        message: messageToLog,
-        additional: additionalToLog,
-        timestamp: this._timestamp()
-      },
-      {
-        headers
-      })
-      .subscribe(
-        (res: any) => this._log(NgxLoggerLevel.TRACE, false, 'Server logging successful', [res]),
-        (error: HttpErrorResponse) => this._log(NgxLoggerLevel.ERROR, false, 'FAILED TO LOG ON SERVER', [error])
-      );
+  public updateConfig(config: LoggerConfig) {
+    this.configService.updateConfig(config);
   }
 
-  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = []): void {
+  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = [], timestamp: string): void {
     switch (level) {
       case NgxLoggerLevel.WARN:
-        console.warn(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.warn(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.ERROR:
-        console.error(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.error(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.INFO:
-        console.info(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.info(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       default:
-        console.log(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.log(`${timestamp} [${Levels[level]}] `, message, ...additional);
     }
   }
 
-  private _log(level: NgxLoggerLevel, logOnServer: boolean, message, additional: any[] = []): void {
-    if (!message) {
-      return;
-    }
-
+  private _prepareMessage(message) {
     try {
       if (message instanceof Error) {
         message = message.stack;
       } else if (typeof message !== 'string') {
         message = JSON.stringify(message, null, 2);
       }
     } catch (e) {
-      additional = [message, ...additional];
+      // additional = [message, ...additional];
       message = 'The provided "message" value could not be parsed with JSON.stringify().';
     }
 
-    // Allow logging on server even if client log level is off
-    if (logOnServer) {
-      this._logOnServer(level, message, additional);
+    return message;
+  }
+
+  private _prepareAdditionalParameters(additional: any[]) {
+    if (additional === null || additional === undefined) {
+      return [];
     }
 
+    const additionalParams: any[] = [];
+
+    for (let i = 0; i < additional.length; i++) {
+      try {
+        if (typeof additional[i] === 'object') {
+          additional[i] = JSON.stringify(additional[i], null, 2);
+        }
+
+        additionalParams.push(additional[i]);
+      } catch (e) {
+        additionalParams.push(`The additional[${i}] value could not be parsed using JSON.stringify().`);
+      }
+    }
+
+    return additionalParams;
+  }
```

Comment:
Personally, I hate `for()` loops, so I am going to offer this as an alternative:
```javascript
private _prepareAdditionalParameters(additional?: any[]) {
  return additional === null || additional === undefined ? [] : additional.map((next, idx) => {
    try {
      return typeof next === 'string' ? next : JSON.stringify(next, null, 2);
    }
    catch (e) {
      return `The additional[${idx}] value could not be parsed using JSON.stringify().`
    }
  });
}
```

Reply:
I'll modify this to use map. However, I think the ternary operator can be difficult to read when more complicated like this. So I'll still keep the null / undefined check separate.

Reply:
also, we have to check for more that just type string here. Really what we want to do is return everything as is, except for objects and arrays (which is handled by typeof 'object')